develop

Opkolopukos opened the pull request.

Opkolopukos committed:
Created method that merges two arrays and writes result in a new txt File.

pvlnsk commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,29 @@
+
+
+import ru.Azarov.mergingfiles.MergeUtil;
```
пакеты с маленькой буквы

pvlnsk commented on src/main/java/ru/Azarov/mergingfiles/MergeUtil.java:
```diff
@@ -0,0 +1,43 @@
+public class MergeUtil {
+    private static final File output = new File("FileMergeOutput.txt");
+
+    public static void merge(int[] first, int[] second, List<Integer> output) {
```
меняй массивы на списки, с массивами негоже работать

pvlnsk replied:
и следующим этапом можешь принять в методе список списков входных данных и один список output оставить, как сейчас есть

Opkolopukos committed:
Changed method arguments to n-Lists.

pvlnsk commented on src/main/java/ru/Azarov/mergingfiles/MergeUtil.java:
```diff
@@ -1,41 +1,42 @@
-    public static void merge(int[] first, int[] second, List<Integer> output) {
-
+    @SafeVarargs
+    public static void merge2Files(List<Integer> output, List<Integer>... lists) {
```
лист листов, немного отличается от массива lists, говорю же уйди от массивов, работай с коллекциями

pvlnsk replied:
чтобы итерироваться по списку есть же foreach

Opkolopukos replied:
> чтобы итерироваться по списку есть же foreach 

Сделал, от массивов на вход избавился. Умный алгоритм сортировки для листа листов (или матриц) пришлось гуглить, конечно. Я про него и говорил, что с ним пришлось погрокаться, тк он показался очень уместным для конкретной задачи.

pvlnsk replied:
Сначала подумал, что такой алгоритм не подойдёт, но он подойдёт. Теперь давай подумаем чем можно заменить списки на входе в алгоритм. И сначала подумаем над требованиям к классу, которым мы заменим списки

Opkolopukos committed:
test:
Changed method arguments to listOfLists.
Added mergeUtils class that includes listOfLists merging method.
Separated Sorter and MergeUtils classes.

Opkolopukos committed:
- Created inputFiles class which processes input files and opens reading streams.
- Changed project format to Gradle (from Maven).
